File: hw/tipantiza/u3/hw27ObserverPattern/ObserverPattern_js/model/Investor.js
const IInvestor = require('./IInvestor');

class Investor extends IInvestor {
    constructor(name) {
        super();
        this._name = name;
    }

    update(stock, args) {
        console.log(`Notified observer ${this._name}`);
        if (typeof args === 'string') {
            console.log(`The symbol of ${stock.getSymbol()} changed to: ${args}`);
        } else if (typeof args === 'number') {
            console.log(`The price of ${stock.getSymbol()} changed to: ${args}`);
        }
    }
}

module.exports = Investor;

File: hw/vizcaino/u3/hw27Observer/hw27ObserverJava/Observer/src/main/java/ec/edu/espe/observer/Investor.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.observer;

/**
 *
 * @author Adrian Vizcaino <The-Softwarriors at ESPE>
 */
public class Investor implements IInvestor {
    private String name;

    public Investor(String name) {
        this.name = name;
    }

    public void update(Stock stock, Object args) {
        System.out.print("Notified observer " + name + " ");
        if (args instanceof String) {
            System.out.println("The symbol of " + stock.getSymbol() + " changed to: " + args);
        } else if (args instanceof Double) {
            System.out.println("The price of " + stock.getSymbol() + " changed to: " + args);
        }
    }
}